packages/core/src/rum/instrumentation/resourceTracking/domain/firstPartyHosts.ts

```diff
@@ -18,9 +23,33 @@ const escapeRegExp = (string: string) => {
     return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
 };
 
-export const firstPartyHostsRegexBuilder = (
+export const firstPartyHostsRegexMapBuilder = (
     firstPartyHosts: FirstPartyHost[]
-): RegExp => {
+): RegexMap => {
+    const hostsMap: Record<PropagatorType, string[]> = {
+        datadog: []
+    };
+
+    firstPartyHosts.forEach(host => {
+        host.propagatorTypes.forEach(propagatorType => {
+            hostsMap[propagatorType].push(host.match);
+        });
+    });
+
+    const regexMap: { regex: RegExp; propagatorType: PropagatorType }[] = [];
+    Object.entries(hostsMap).forEach(([propagatorType, hosts]) => {
+        if (hosts.length > 0) {
+            regexMap.push({
+                propagatorType: propagatorType as PropagatorType,
+                regex: firstPartyHostsRegexBuilder(hosts)
+            });
+        }
+    });
+
+    return regexMap;
+};
```

Comment:
We have 3 different data structures in this feature, I'm afraid it's going to become confusing at some point.
In the configuration, we match the data structure from the `browser-sdk`:
```
[{ match: "api.com", propagatorTypes: [PropagatorTypes.DATADOG] }]
```

In the native side, we have to match the native configuration, which basically turns the array into a dictionary with the `match` as key and `propagatorTypes` as value: 

```
{ "api.com": [DATADOG] }
```

And here to be able to build the regex, we inverse the mapping. Instead of mapping by host, we map by propagator type:

```
{ "datadog": ["api.com"] }
```

We have to use the native SDK structure at one point.
I believe it's better to use the same configuration as browser for 2 reasons:
1. We are more likely to have web devs also doing the RN instrumentation (rather than Android/iOS dev doing RN)
2. Eventually we'll want to support browser in RN

Now, if we want to get rid of this type, we have to re-think the implementation of the host detection. I believe we can have the discussion now, but in any case I would prefer to implement it in a separate PR to really focus on it.